src/js/components/TaskDetailComponent.jsx

```diff
@@ -41,10 +42,33 @@ var TaskDetailComponent = React.createClass({
   getTaskEndpoints: function () {
     var props = this.props;
     var task = props.task;
-    if (task.ports == null || task.ports.length === 0) {
+    var app = AppsStore.getCurrentApp(props.appId);
+
+    if ((task.ports == null || task.ports.length === 0) &&
+        (task.ipAddresses == null || task.ipAddresses.length === 0)) {
       return (<dd>None</dd>);
     }
 
+    if (app.ipAddress != null &&
```

Comment:
Yeah I thought about it... but decided to keep it self-contained. Should I refactor?